dts/bindings/pinctrl/nxp,mcux-rt-pinctrl.yaml

```diff
@@ -0,0 +1,132 @@
+      nxp,speed = <3>;
+      slew-rate = <1>;
+  }
+
+  This will select GPIO_AD_B0_12 for use as SCL on LPI2C4, and enable
+  the 100K pull up resistor on the pin. It will also increase the slew rate
+  and drive strength to their maximum settings. Note that the soc level pinctrl
+  header file can be examined to find the possible pinmux options.
+
+  Here are the affects of each property on the IOMUXC SW_PAD_CTL register:
+
+  input-schmitt-enable: HYS=1
+  drive-open-drain: ODE=1
+  input-enable: SION=1 (in SW_MUX_CTL_PAD register)
+  bias-bus-hold: PKE=1, PUE=0
+  bias-pull-down: PKE=1, PUE=1, PUS=0
+  bias-pull-up: PKE=1, PUE=1, PUS=<prop_value>
+  slew-rate: SRE=<prop_value>
+  drive-strength: DSE=<prop_value>
+  nxp,speed: SPEED=<prop_value>
+
+
+
+compatible: "nxp,mcux-rt-pinctrl"
+
+include:
+    - name: base.yaml
+    - name: pincfg-node-group.yaml
+      child-binding:
+        child-binding:
+          property-allowlist:
+            - input-schmitt-enable
+            - drive-open-drain
+            - input-enable
+            - bias-bus-hold
+            - bias-pull-down
+
+properties:
+    reg:
+      required: true
+
+child-binding:
+  description: MCUX RT pin controller pin group
+  child-binding:
+    description: |
+      MCUX RT pin controller pin configuration node.
+    properties:
+      pinmux:
+        required: true
+        type: array
+        description: |
+          An array of values defining the configuration register for a given
+          pinmux mode. See fsl_iomuxc.h in the NXP HAL for a list.
+          The array takes the following format:
+          mux_register, mux_value, input_register, input_daisy, config_register
+      drive-strength:
+        required: false
+        type: int
+        enum:
+          - 0
+          - 1
+          - 2
+          - 3
+          - 4
+          - 5
+          - 6
+          - 7
+        description: |
+          Pin output drive strength. Sets the DSE field in the IOMUXC peripheral.
+          000 DSE_0_output_driver_disabled_ — output driver disabled
+          001 DSE_1_R0_1 — 157 Ohm impedance @3.3V, 260 Ohm impedance @1.8V
+          010 DSE_2_R0_2 — 78 Ohm @3.3V, 130 Ohm @1.8V
+          011 DSE_3_R0_3 — 53 Ohm @3.3V, 88 Ohm @1.8V
+          100 DSE_4_R0_4 — 39 Ohm @3.3V, 65 Ohm @1.8V
+          101 DSE_5_R0_5 — 32 Ohm @3.3V, 52 Ohm @1.8V
+          110 DSE_6_R0_6 — 32 Ohm @3.3V, 43 Ohm @1.8V
+          111 DSE_7_R0_7 — 26 Ohm @3.3V, 37 Ohm @1.8V
+      bias-pull-up:
+        required: false
+        type: int
+        enum:
+          - 0
+          - 1
+          - 2
+          - 3
+        description: |
+          Select the value of the pull up resistor present on this pin
+          Corresponds to the PUS field in the IOMUXC peripheral
+          00 Unused- no change will be applied to pin
+          01 PUS_1_47K_Ohm_Pull_Up — 47K Ohm Pull Up
+          10 PUS_2_100K_Ohm_Pull_Up — 100K Ohm Pull Up
+          11 PUS_3_22K_Ohm_Pull_Up — 22K Ohm Pull Up
```

Comment:
I'd do here instead (to follow standard properties as much as possible):

```
      # note: bias-pull-up added to the allowlist
      bias-pull-up-value:
        required: false
        type: int
        enum:
          - "unused"
          - "47k"
          - "100k"
          - "22k"
        description: |
          Select the value of the pull up resistor.
```

Then do this in boards:

```
bias-pull-up;
bias-pull-up-value = "100k";
```

and to store value:

```
.pullup = DT_PROP(node_id, bias_pull_up) * DT_ENUM_IDX(node_id, bias_pull_up_value),
```

---

Or maybe:
```c
// include/dt-bindings/pinctrl/nxp/mcux-rt-pinctrl.h

#define MCUX_PULL_47K 0x1
#define MCUX_PULL_100K 0x2
...
```

```
      bias-pull-up-value:
        required: false
        type: int
        description: |
          Select the value of the pull up resistor (use MCUX_PULL_XXX pre-defined vaues).
```

```
bias-pull-up;
bias-pull-up-value = <MCUX_PULL_100K>;
```

```
.pullup = DT_PROP(node_id, bias_pull_up) * DT_PROP(node_id, bias_pull_up_value),
```

Reply:
This makes sense. I'll switch the pull up implementation to use a format like this, and keep the `bias_pull_up` property standard.